### ISSUE TITLE
AttributeError: type object 'Task' has no attribute 'all_tasks' for python3.9

### ISSUE THREAD

**Issue by unkcpz** [aiidateam/aiida-core#4594]
- [x] [AiiDA Troubleshooting Documentation](https://aiida.readthedocs.io/projects/aiida-core/en/latest/intro/troubleshooting.html)
- [x] [AiiDA Users Forum](https://groups.google.com/forum/#!forum/aiidausers)

### Describe the bug
OK, here is a issue after I start using asyncio version with PY3.9. The  `asyncio.Task.all_tasks()` and `asyncio.Task.current_task()` is deprecated in PY3.9 but needed for PY3.6. I think before we dropping py3.6 we need to add a python version check (`sys.version_info`) to support both of them. 

OR any other workaround exist?

```
Traceback (most recent call last):                                                                                        
File "/data/CONDA_ENV/aiida-testing/lib/python3.9/asyncio/tasks.py", line 256, in __step                              
    result = coro.send(None)                                                                                              
File "/home/unkcpz/pyProject/aiida_core/aiida/engine/daemon/runner.py", line 39, in shutdown                          
    tasks = [t for t in asyncio.Task.all_tasks() if t is not asyncio.Task.current_task()]
AttributeError: type object 'Task' has no attribute 'all_tasks'
```

### Steps to reproduce

Just stop the daemon.

### Your environment

- Operating system Linux:
- Python version 3.9:
- aiida-core version [develop-branch with asyncio]:

Other relevant software versions, e.g. Postres & RabbitMQ

**Comment by unkcpz** [comment]
BTW, this issue also causes the daemon failed to stop. Thus this is an emergence issue to be fixed

**Comment by sphuber** [comment]
I am a bit confused because according to [this PR](https://github.com/python/cpython/commit/004e64e8059fe68a72890314673282f2e60d5ce1) it should have been removed in Python 3.9 (as you also seem to claim) but then it seems to have only been merged into their `v3.0.0a` tags. Maybe their Github tags are not synchronized with the actual releases but that would seem odd. Also on the original [issue tracker](https://bugs.python.org/issue40967) there was discussion of moving the removal from 3.9 to 3.10

**Comment by sphuber** [comment]
And why wasn't this caught by the tests? The daemon is [stopped in the test workflow](https://github.com/aiidateam/aiida-core/blob/716a1d8f6801bd9ba72162a73497e1085a64bc52/.github/workflows/tests.sh#L23). And [in this action](https://github.com/aiidateam/aiida-core/runs/1465398801?check_suite_focus=true#step:10:91) it was tested against Python 3.9 and the daemon was stopped just fine.

Never mind, I see now that the call to stop the daemon actually timed out. Maybe there is way we can ensure here that the daemon is no longer running.

**Comment by unkcpz** [comment]
> And why wasn't this caught by the tests? 

It is `Timeout` and the daemon then is somehow being stopped successfully this will not be caught by the test case.

So, for the time being, we add a version check to workaround this in PY3.9?

**Comment by sphuber** [comment]
@unkcpz yes, the solution is a simple conditional on the version to import the correct method, but I would first want to add a test so this is caught in the future. I already started [working on a branch](https://github.com/sphuber/aiida-core/runs/1473247514) so I can take care of this. I added a check for the daemon being successfully shutdown, except even for Python 3.6 the test is failing. Don't know why yet
